extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +152,215 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(win, transportOptions, opt_processResponse) {
+    const frameUrl = dev().assertString(transportOptions['iframe'],
+      'Cross-domain frame parameters missing ${this.type_}');
+    const frameData = this.useExistingOrCreateCrossDomainIframe_(
+      win, frameUrl, transportOptions['extraData']);
+    const iframeMessagingClient = frameData.iframeMessagingClient;
+    iframeMessagingClient.registerCallback(
+      AMP_ANALYTICS_3P_MESSAGE_TYPE.READY, () => {
+        iframeMessagingClient.setHostWindow(frameData.frame.contentWindow);
+        frameData.newCreativeMessageQueue.setIsReady();
```

Comment:
Hmm...the spec (https://html.spec.whatwg.org/multipage/web-messaging.html#dom-window-postmessage) includes the following language:
"7. Queue a task" - this implies a FIFO nature, but doesn't spell it out explicitly
"7.6 [...] maintaining their relative order" - we're good here
"7.7 Fire an event" - drilling down into this, and then into step 5 "dispatching", says "append a pointer to the event listener to listeners" which again strongly implies ordering.
...but I agree this doesn't necessarily guarantee.
If we get an event before a new creative message, do you prefer to 1) go ahead and send the event to the creative, which would not be able to access the extraData at that point since it hasn't arrived yet, or 2) queue the event until we receive the new creative message?